File: deploy_codesandbox.js
const fs = require('fs');
const https = require('https');

const htmlContent = fs.readFileSync('dist/index.html', 'utf8');

const data = JSON.stringify({
  files: {
    "package.json": {
      "content": {
        "main": "index.html",
        "scripts": { "start": "serve ." },
        "dependencies": { "serve": "^13.0.0" }
      }
    },
    "index.html": {
      "content": htmlContent
    }
  }
});

const options = {
  hostname: 'codesandbox.io',
  path: '/api/v1/sandboxes/define?json=1',
  method: 'POST',
  headers: {
    'Content-Type': 'application/json',
    'Content-Length': Buffer.byteLength(data)
  }
};

const req = https.request(options, res => {
  let body = '';
  res.on('data', d => body += d);
  res.on('end', () => {
    console.log('Response:', body);
  });
});

req.on('error', e => console.error(e));
req.write(data);
req.end();

File: deploy_codesandbox.py
import json
import urllib.request

with open('dist/index.html', 'r', encoding='utf-8') as f:
    html_content = f.read()

data = {
    "files": {
        "package.json": {
            "content": {
                "name": "beta-club-tracker",
                "version": "1.0.0",
                "main": "index.html",
                "scripts": {
                    "start": "serve ."
                },
                "dependencies": {
                    "serve": "^13.0.0"
                }
            }
        },
        "index.html": {
            "content": html_content
        }
    }
}

req = urllib.request.Request(
    'https://codesandbox.io/api/v1/sandboxes/define?json=1',
    data=json.dumps(data).encode('utf-8'),
    headers={'Content-Type': 'application/json', 'Accept': 'application/json'}
)

try:
    with urllib.request.urlopen(req) as response:
        print(response.read().decode('utf-8'))
except Exception as e:
    print(e)
